### ISSUE TITLE
Use ! function for not logic

### ISSUE THREAD

**Issue by mtfishman** [ITensor/ITensors.jl#470]
As an alias for the `not` function, we can use the function `!` (I just realized it was available as a function).

For example, to prime all indices in ITensor `A` besides `i`, we could use the syntax: `prime(A; inds = !i)`. Right now this gives:
```julia
julia> i = Index(2, "i"); j = Index(2, "j");

julia> A = randomITensor(i, j);

julia> prime(A, inds = !i)
ERROR: MethodError: no method matching !(::Index{Int64})
[...]
```
We can also use the same syntax for not priming indices with a certain tag, such as:
```julia
julia> prime(A, tags = !"i")
ERROR: MethodError: no method matching !(::String)
[...]
```
but we should be careful about that, since both the `!` function and the `String` type are not defined by ITensors.jl. Therefore, it is type piracy, and other packages may have a definition for `!(::String)`.

**Comment by mtfishman** [comment]
Another interesting point I hadn't noticed is that Julia automatically defines the operation `!(::Function)` to negate the boolean result of the function. So that means the following already works:
```julia
julia> prime(!hastags("i"), A)
ITensor ord=2 (dim=2|id=41|"i") (dim=2|id=812|"j")'
NDTensors.Dense{Float64,Array{Float64,1}}
```
which could make the implementation of the `not` logic in ITensors.jl simpler.

**Comment by mtfishman** [comment]
Closed by #471.
